tests/substrate_tests/base58_encoding.rs

```diff
@@ -0,0 +1,38 @@
+use crate::build_solidity;
+use base58::ToBase58;
+use parity_scale_codec::{Decode, Encode};
+
+#[test]
+fn print_addresses() {
+    #[derive(Debug, PartialEq, Eq, Encode, Decode)]
+    struct Address([u8; 32]);
+
+    let mut runtime = build_solidity(
+        r#"
+        contract Base58 {
+    function print_this(address addr) pure public {
+        print("{}".format(addr));
+    }
+}
+        "#,
+    );
+
+    runtime.constructor(0, Vec::new());
+
+    let addr = Address([
+        0x7d, 0x58, 0x39, 0xe2, 0x4a, 0xca, 0xda, 0x33, 0x8c, 0x25, 0x76, 0x43, 0xa7, 0xd2, 0xe0,
+        0x25, 0x45, 0x3f, 0x77, 0xd0, 0x58, 0xb8, 0x33, 0x5c, 0x1c, 0x37, 0x91, 0xbc, 0x67, 0x42,
+        0xb3, 0x20,
+    ]);
+    runtime.function("print_this", addr.encode());
+    assert!(runtime.printbuf.contains(addr.0.to_base58().as_str()));
```

Comment:
This is not how address are base58 encoded on Substrate. There is a prefix of some sort, and it uses the blake2 hash: https://github.com/hyperledger/solang/blob/main/src/sema/expression/literals.rs#L211

Reply:
Yes, Substrate uses their so called [SS58 address format](https://docs.substrate.io/reference/address-formats/).

Reply:
Ok, I return the formatting in Substrate to hex.